pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -178,6 +183,12 @@ func (r *ReconcileConnection) Reconcile(request reconcile.Request) (reconcile.Re
 		tunnelSecret,
 	)
 
+	if err = imageset.ApplyImageSet(ctx, r.Client, variant, component); err != nil {
```

Comment:
I'm just skimming, but why did you decide to go with a pattern of modifying the component after creating it rather than passing the image set to the constructor?

Reply:
I think the main reason was I wanted it to be obvious that a Component needs to have the ResolveImages functionality.
(I wasn't confident on this pattern and is why I put out a draft PR of this last week.)

Reply:
I don't think I really have a problem with it, it just wasn't my first thought. I was imagining that we'd just add an argument to `GetReference` and that would be obvious enough that an imageset is required anywhere that an image is used.

Reply:
Hmmm, my reasoning for not doing that was because of where we were calling GetReference and changing it like that could cause an error and didn't want to change Objects signature. So I wanted that resolution to happen sooner (not in Objects()), but I guess I've changed that behavior already by moving those calls into ResolveImages.